Manage.c

```diff
@@ -278,10 +278,8 @@ int Manage_stop(char *devname, int fd, int verbose, int will_retry)
 		 */
 		mds = mdstat_read(0, 0);
 		for (m = mds; m; m = m->next)
-			if (m->metadata_version &&
-			    strncmp(m->metadata_version, "external:", 9)==0 &&
-			    metadata_container_matches(m->metadata_version+9,
-						       devnm)) {
+			if (is_mdstat_ent_external(m) &&
+			    metadata_container_matches(m->metadata_version + 9, devnm)) {
```

Comment:
This patch might be an opportunity to get rid of "+9" magic number and use a define (like VERSION_SHIFT or smth.), but I leave it up to you.

Reply:
I tried.. it is too big task. I  requires to rework a lot of code and it is not simple sometimes, I need to be extra careful (and I may not be able to test everything.

Let me finish this for now!

Reply:
Simple replacement of `+ 9` by `+ EXTERNAL_INDICATOR_LEN` makes it only worse (I personally don't like it). The right way to do it is to propose separate family to handle `metadata version`. I tried  once and I ended with 5 inlines and it was not enough, it will lead me to rework almost each mdstat usage. This must be done separately.
This time, I want to end mdstat lists rework and I don't want to complicate this task more. It is 3th story started by me on on ``--set-faulty`` and ``--remove`` problem.

Also, to propose "correct" naming it all must be done together. I had a problem how to name `/md127/1` substring extracted from `metadata_version`. 

Thanks for this suggestion but I can't for now :(

Reply:
Ack :)